includes/data-stores/class-wc-admin-reports-variations-data-store.php

```diff
@@ -78,11 +78,41 @@ public function __construct() {
 		$this->report_columns['orders_count'] = str_replace( 'order_id', $table_name . '.order_id', $this->report_columns['orders_count'] );
 	}
 
+	/**
+	 * Fills ORDER BY clause of SQL request based on user supplied parameters.
+	 *
+	 * @param array $query_args Parameters supplied by the user.
+	 *
+	 * @return array
+	 */
+	protected function get_order_by_sql_params( $query_args ) {
+		global $wpdb;
+		$order_product_lookup_table = $wpdb->prefix . self::TABLE_NAME;
+
+		$sql_query['order_by_clause'] = '';
+		if ( isset( $query_args['orderby'] ) ) {
+			$sql_query['order_by_clause'] = $this->normalize_order_by( $query_args['orderby'] );
+		}
+
+		if ( 'postmeta.meta_value' === $sql_query['order_by_clause'] ) {
+			$sql_query['from_clause'] .= " JOIN {$wpdb->prefix}postmeta AS postmeta ON {$order_product_lookup_table}.variation_id = postmeta.post_id AND postmeta.meta_key = '_sku'";
```

Comment:
I have been testing this when a comparison is active and couldn't get it working. You can reproduce it selecting a variable product, then selecting a _Comparison_ filter and selecting some of them:
![image](https://user-images.githubusercontent.com/3616980/54989150-37c8e400-4fb8-11e9-924a-357932088332.png)

This is a bit complex because the SQL command changes depending if a comparison is active or not. You can use `sku` sorting in the products table as a reference in case it helps:
https://github.com/woocommerce/woocommerce-admin/blob/master/includes/data-stores/class-wc-admin-reports-products-data-store.php#L135
Basically, you will have to make that join to be in the inner query when there is no comparison and to be in the outer query when there is a comparison.

I think the product endpoint is quite similar to this one, so taking a look at the file I linked, should be helpful. :+1:

Reply:
@Aljullu , Thanks for reviewing these changes and sharing your thoughts. I am looking for this solution till that what I suggest is that let's create a separate issue for this and merge these changes so that it can be less chance of conflicts.

Reply:
@ronakganatra9 thank you for addressing the other issues!

I would say it's better to wait until the feature is complete before merging it into master. Otherwise, users might try sorting by `sku` with an active comparison and that would make Woocommerce Admin to crash.

As far as I know, there is nobody currently working in the same files you modified in this PR, so I don't expect you to end up with many conflicts. In any case, if you have issues with that, we can help in solving them.

Does that sound good?

Reply:
@Aljullu , Comparision is not available on products analytics page so it is not occurring the issue there. But in variation page of single product, there is a comparison option available so I recommend to push these changes or create a different issue for that because that is a complex issue.

Reply:
Thanks for your patience here @ronakganatra9, there were so many things going on last week I didn't have time to take another look at this until now.

I created a follow-up PR that will fix the issue with the comparison, so you don't have to worry about that, and we can proceed merging this one. :slightly_smiling_face: 

Thank you again for your help!

Reply:
Thank you so much @Aljullu for this.